src/sortStudents.ts

```diff
@@ -1,16 +1,69 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+function averageGrade(grades: number[]) : number {
+  return grades.reduce((a: number, b: number) => a + b, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudent: Student[] = [...students];
```

Comment:
```suggestion
  const copyOfStudents: Student[] = [...students];
```
It's a copy of all students, not just one